web-client/src/types.ts

```diff
@@ -0,0 +1,3 @@
+export type DataReferenceType = firebase.firestore.DocumentReference<
```

Comment:
This should be removed and similar type should be added to the types for individual collections, like PostRefType,  UserRefType, etc.